hera_sim/simulate.py

```diff
@@ -276,15 +282,34 @@ def write(self, filename, save_format="uvh5", save_seeds=True, **kwargs):
         # TODO: docstring
         """
         """
+        if save_seeds:
+            # Update the extra_keywords attribute of the underlying UVData
+            # object to contain the seed information.
+            seed_dict = {}
+            for component, seeds in self._seeds.items():
+                if len(seeds) == 1:
+                    seed = list(seeds.values())[0]
+                    key = "_".join([component, "seed"])
+                    seed_dict[key] = seed
+                else:
+                    # This should only be raised for seeding by redundancy.
+                    # Each redundant group is denoted by the *first* baseline
+                    # integer for the particular redundant group. See the
+                    # _generate_redundant_seeds method for reference.
+                    for bl_int, seed in seeds.items():
+                        key = "_".join([component, "seed", str(bl_int)])
+                        seed_dict[key] = seed
+            
+            # Now actually update the extra_keywords dictionary.
+            self.data.extra_keywords.update(seed_dict)
         try:
             getattr(self.data, "write_%s" % save_format)(filename, **kwargs)
         except AttributeError:
             msg = "The save_format must correspond to a write method in UVData."
             raise ValueError(msg)
-        if save_seeds:
-            seed_file = os.path.splitext(filename)[0] + "_seeds"
-            np.save(seed_file, self._seeds)
 
+    # XXX with the new version of the CLI, this should not need to be wrapped 
```

Comment:
Is this true? Can it be tested?

Reply:
It's definitely true now. The previous version of the CLI basically parsed the passed configuration file to generate a configuration to be passed to `Simulator.run_sim`, so the generator-like behavior was needed in order to actually extract the individual data products. The new version of the CLI just loops over the parsed configuration and applies effects using the `Simulator.add` method.

Reply:
So I updated this comment a bit with something I thought was worth considering. The `_generator_to_list` decorator allows the user to receive a list with all (or a subset of) the different simulation components... this might be a handy feature. I kind of want to scrap it and require that the user uses `Simulator.get` to retrieve simulation components, especially because the implementation used here is prone to the return value being confusing.

Do you have any thoughts on this?

Reply:
"There should be one-- and preferably only one --obvious way to do it."

I think making the user be explicit and use `Simulator.get` is entirely reasonable, unless that can get non-obvious (perhaps if they potentially don't know what they've simulated?).

Reply:
After thinking about this in the context of how I've used `hera_sim` for validation tests, I think I would rather scrap the option to have `Simulator.run_sim` return anything and require that end users who want particular simulation components either explicitly simulate effects one-by-one (and use the `ret_vis` keyword) or run the full simulation and use `Simulator.get`.